tsl/test/expected/split_chunk.out

```diff
@@ -382,27 +493,380 @@ select create_hypertable('splitme_md', 'time', 'device', 2, chunk_time_interval
+truncate chunk_data_after_split;
+insert into chunk_data_after_split select * from _timescaledb_internal._hyper_1_3_chunk;
+insert into chunk_data_after_split select * from _timescaledb_internal._hyper_1_16_chunk;
+truncate chunk_summary_after_split;
+insert into chunk_summary_after_split
+select
+    count(*),
+    sum(device) as device_sum,
+    sum(location) as location_sum,
+    round(sum(temp)::numeric, 5) as temp_sum
+    from chunk_data_after_split;
+-- Compare summaries before and after split
+select * from chunk_summary_before_split;
+ count | device_sum | location_sum |   temp_sum   
+-------+------------+--------------+--------------
+ 29525 |     162733 |       169511 | 604005.96562
+(1 row)
+
+select * from chunk_summary_after_split;
+ count | device_sum | location_sum |   temp_sum   
+-------+------------+--------------+--------------
+ 29525 |     162733 |       169511 | 604005.96562
+(1 row)
+
+-- Show the summary for each new chunk
+select
+    count(*),
+    sum(device) as device_sum,
+    sum(location) as location_sum,
+    round(sum(temp)::numeric, 5) as temp_sum
+    from _timescaledb_internal._hyper_1_3_chunk;
+ count | device_sum | location_sum |   temp_sum   
+-------+------------+--------------+--------------
+ 28801 |     158662 |       165240 | 589129.85016
+(1 row)
+
+select
+    count(*),
+    sum(device) as device_sum,
+    sum(location) as location_sum,
+    round(sum(temp)::numeric, 5) as temp_sum
+    from _timescaledb_internal._hyper_1_16_chunk;
+ count | device_sum | location_sum |  temp_sum   
+-------+------------+--------------+-------------
+   724 |       4071 |         4271 | 14876.11546
+(1 row)
+
+select chunk_name, range_start, range_end, is_compressed
+from timescaledb_information.chunks
+where hypertable_name = 'splitme'
+order by chunk_name, range_start, range_end;
+    chunk_name     |         range_start          |          range_end           | is_compressed 
+-------------------+------------------------------+------------------------------+---------------
+ _hyper_1_13_chunk | Thu Jan 04 13:40:30 2024 PST | Fri Jan 05 16:00:00 2024 PST | t
+ _hyper_1_16_chunk | Tue Jan 09 00:00:00 2024 PST | Wed Jan 10 16:00:00 2024 PST | t
+ _hyper_1_1_chunk  | Wed Jan 03 16:00:00 2024 PST | Thu Jan 04 00:00:00 2024 PST | f
+ _hyper_1_2_chunk  | Thu Jan 04 00:00:00 2024 PST | Thu Jan 04 13:40:30 2024 PST | t
+ _hyper_1_3_chunk  | Sun Jan 07 08:00:00 2024 PST | Tue Jan 09 00:00:00 2024 PST | t
+ _hyper_1_7_chunk  | Wed Dec 27 16:00:00 2023 PST | Wed Jan 03 16:00:00 2024 PST | f
+ _hyper_1_8_chunk  | Fri Jan 05 16:00:00 2024 PST | Sun Jan 07 08:00:00 2024 PST | f
+(7 rows)
+
+select compress_relid from _timescaledb_catalog.compression_settings
+where relid = '_timescaledb_internal._hyper_1_16_chunk'::regclass \gset
+select count(*) from :compress_relid;
+ count 
+-------
+    11
+(1 row)
+
+-- Split compressed chunk in a way that leaves no compressed data in
+-- one of the chunks
```

Comment:
Are you testing both combinations? That is, moving (all) compressed data to the new chunk and moving (all) uncompressed data to the new chunk.

Reply:
There are tests to split so that one chunk gets all compressed data and the other chunk gets non-compressed data. But I am not sure the same test exists for both "left" and "right"... Do you think that is necessary? The test is already quite big and hard to read so not convinced of the value of adding this since it is more or less covered by all other combos.